pyvista/core/pointset.py

```diff
@@ -142,50 +197,74 @@ class PolyData(_vtk.vtkPolyData, PointSet, PolyDataFilters):
                 '.stl': _vtk.vtkSTLWriter,
                 '.vtk': _vtk.vtkPolyDataWriter}
 
-    def __init__(self, *args, **kwargs) -> None:
+    def __init__(self, var_inp=None, faces=None, n_faces=None, lines=None,
+                 n_lines=None, deep=False) -> None:
         """Initialize the polydata."""
+        points = None
+        local_parms = locals()
         super().__init__()
 
-        deep = kwargs.pop('deep', False)
+        # allow empty input
+        if var_inp is None:
+            return
```

Comment:
I'm split about this.  We basically allow either a filename, points, or points and faces and/or lines.  I think this seems like the easiest way of implementing this.  The check is quite complex as it is.

Reply:
I was wondering about `pv.PolyData(faces=faces, lines=lines)` which would keep `var_inp = None`. But now I think this is nonsense, because you need the points, and you can only pass points as the first (positional) argument. So I think this is fine.